src/shapes/image.cpp

```diff
@@ -14,6 +14,9 @@ void Image::draw(Renderer* renderer)
 	}
 	renderer->save();
 
+	// Clip and tell it we're already saved (avoids extra save op).
+	clip(renderer, false);
```

Comment:
lgtm, but couldn't we also have just *not* called save on line 15?

Reply:
Yes! We could've but we'd have had to track whether or not clip did save. See how shape.cpp does it. Clip doesn't save if there was no clipping applied to this Drawable, so other Drawables can not save at all if they don't need to.

I think your suggestion keeps clip simpler, which is a good idea and normalizes how we use it..